drivers/wireless/ieee80211/bcm43xxx/bcmf_chip_43362.c

```diff
@@ -45,39 +45,41 @@ extern const uint8_t bcm43362_firmware_image[];
 extern const unsigned int bcm43362_firmware_image_len;
 #endif
 
-const struct bcmf_sdio_chip bcmf_43362_config_sdio =
+struct bcmf_sdio_chip bcmf_43362_config_sdio;
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void bcmf_43362_init(struct bcmf_sdio_chip *conf)
 {
+  uint32_t *core = conf->core_base;
+
   /* General chip stats */
 
-  .ram_base = 0,
-  .ram_size = 0x3c000,
+  conf->ram_base = 0;
+  conf->ram_size = 0x3c000;
 
   /* Backplane architecture */
 
-  .core_base =
-  {
-    [CHIPCOMMON_CORE_ID]  = 0x18000000,  /* Chipcommon core register base   */
-    [DOT11MAC_CORE_ID]    = 0x18001000,  /* dot11mac core register base     */
-    [SDIOD_CORE_ID]       = 0x18002000,  /* SDIOD Device core register base */
-    [WLAN_ARMCM3_CORE_ID] = 0x18003000 + /* ARMCM3 core register base       */
-                            WRAPPER_REGISTER_OFFSET,
-    [SOCSRAM_CORE_ID]     = 0x18004000 + /* SOCSRAM core register base      */
-                            WRAPPER_REGISTER_OFFSET
-  },
+  core[CHIPCOMMON_CORE_ID]  = 0x18000000;  /* Chipcommon core register base   */
+  core[DOT11MAC_CORE_ID]    = 0x18001000;  /* dot11mac core register base     */
+  core[SDIOD_CORE_ID]       = 0x18002000;  /* SDIOD Device core register base */
+  core[WLAN_ARMCM3_CORE_ID] = 0x18003000   /* ARMCM3 core register base       */
+                            + WRAPPER_REGISTER_OFFSET;
+  core[SOCSRAM_CORE_ID]     = 0x18004000   /* SOCSRAM core register base      */
+                            + WRAPPER_REGISTER_OFFSET;
 
   /* Firmware images */
 
   /* TODO find something smarter than using image_len references */
 
-  .nvram_image         = (FAR uint8_t *)bcm43362_nvram_image,
-  .nvram_image_size    = (FAR unsigned int *)&bcm43362_nvram_image_len,
+  conf->nvram_image         = (FAR uint8_t *)bcm43362_nvram_image;
```

Comment:
move after ram_base/ram_size

Reply:
I have updated the patch (disregarding C89 so far).